### PR TITLE
Made HandlerController package private.

### PR DESCRIPTION
**Pull request by cmelchior**
Part of #1594. Merge all in one release. 

@realm/java 

Final commit note:
-----
Title: Remove HandlerController from the public API

HandlerController was made public by mistake. This is an class only used by internal API's, so shouldn't be public.

Most likely it was public due to a testing class (HandlerProxy) in a separate package needing access to it. HandlerProxy was instead moved to `io.realm` to work around this as keeping the public API clean has priority.

### CODE REVIEW

**Comment by kneth** [comment]
:+1:

**Comment by beeender** [comment]
:+1:

File: realm/realm-library/src/main/java/io/realm/HandlerController.java [hunk 1]
```diff
@@ -38,12 +38,12 @@
 import io.realm.internal.SharedGroup;
 import io.realm.internal.async.BadVersionException;
 import io.realm.internal.async.QueryUpdateTask;
-import io.realm.internal.log.RealmLog;
+import io.realm.internal.log.RealmLog;s
```
Comment (stk1m1) [inline realm/realm-library/src/main/java/io/realm/HandlerController.java]: `s` at the end.

**Comment by stk1m1** [comment]
👍

**Comment by cmelchior** [comment]
Retest this please

**Comment by cmelchior** [comment]
CI: retest this

**Pull request merged**